src/components/organisms/Badges/Badges.tsx

```diff
@@ -1,77 +1,85 @@
-import React, { useMemo } from "react";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
 import { User } from "@bugsnag/js";
 
 import { Link } from "react-router-dom";
-import { useFirestoreConnect } from "react-redux-firebase";
-
-import { useConnectRelatedVenues } from "hooks/useConnectRelatedVenues";
-import { useSelector } from "hooks/useSelector";
+import { useFirestore } from "react-redux-firebase";
 
 import { PLAYA_VENUE_NAME } from "settings";
 
 import { CampRoomData } from "types/CampRoomData";
 import { isVenueWithRooms } from "types/CampVenue";
-import { AnyVenue } from "types/Firestore";
+import { AnyVenue, UserVisit } from "types/Firestore";
 import { Venue } from "types/Venue";
 
 import { WithId } from "utils/id";
-import {
-  orderedVenuesSelector,
-  userModalVisitsSelector,
-} from "utils/selectors";
 import { venueInsideUrl, venuePreviewUrl } from "utils/url";
 import { notEmpty } from "utils/types";
 
+import { BadgeImage } from "./BadgeImage";
+
 export const Badges: React.FC<{
   user: WithId<User>;
   currentVenue: WithId<Venue>;
 }> = ({ user, currentVenue }) => {
-  useFirestoreConnect([
-    {
-      collection: "users",
-      doc: user.id,
-      subcollections: [{ collection: "visits" }],
-      storeAs: "userModalVisits",
-    },
-  ]);
-  useFirestoreConnect("venues");
-  const visits = useSelector(userModalVisitsSelector);
-  const venues = useSelector(orderedVenuesSelector);
-
-  const { relatedVenues } = useConnectRelatedVenues({
-    venueId: currentVenue.id,
-    withEvents: false,
-  });
-  const relatedVenueNames = relatedVenues.map((venue) => venue.name);
+  const [visits, setVisits] = useState<WithId<UserVisit>[]>([]);
+  const [venues, setVenues] = useState<WithId<AnyVenue>[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
```

Comment:
prefer `isLoading` style
```suggestion
  const [isLoading, setLoading] = useState<boolean>(false);
```